### PR TITLE
557 - Fix createdAt override in publishGrant

### PR DESCRIPTION
**Pull request by ghostffcode**

### CODE REVIEW

**Review by codenamejason — APPROVED** [review]
good catch ser 🔥

File: client/src/actions/newGrant.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const grantCreated = ({
 });
 
 export const publishGrant =
-  (grantId?: string) =>
+  (grantId?: string, updating?: boolean) =>
```
Comment (DanieleSalatti) [inline client/src/actions/newGrant.ts]: `updating` is redundant, we can figure out if we are updating or creating based on whether `grantId` is undefined or not

File: client/src/actions/newGrant.ts [hunk 2]
```diff
@@ -92,7 +92,9 @@ export const publishGrant =
     }
 
     application.credentials = formCredentials;
-    application.createdAt = Date.now();
+    if (!updating) {
```
Comment (DanieleSalatti) [inline client/src/actions/newGrant.ts]: This could be changed to:

`if (!grantId) {`

File: client/src/actions/newGrant.ts (hunk 1 above)
Comment (ghostffcode) [reply client/src/actions/newGrant.ts]: Thanks for pointing this out

File: client/src/actions/newGrant.ts (hunk 2 above)
Comment (ghostffcode) [reply client/src/actions/newGrant.ts]: Thanks, it's more robust now

**Pull request merged**